.github/gradle/gradle.gradle

```diff
@@ -1,41 +1,131 @@
+import groovy.json.JsonBuilder
+import groovy.transform.CompileStatic
+
 if (!project.hasProperty('githubCiTesting') && System.env['GITHUB_OUTPUT'] == null) {
     // If the property is not set and we are not running in the CI environment then we will do nothing
     return
 }
 
-static def outputUnitTest(Project project, Task test) {
-    project.getLogger().lifecycle("<< TEST >>${test.path}")
+static Map<String, String> removeCommonPrefix(List<String> list) {
+    if (list.size() == 0) {
+        return new HashMap<String, String>()
+    }
+
+    def result = new HashMap<String, String>()
+
+    if (list.size() == 1) {
+        System.out.println("Only one item in the list: ${list}")
+        def sections = list[0].split("\\.")
+        result.put(list[0], sections.last())
+        return result
+    }
+
+    Collections.sort(list, { String a, String b -> a.length() - b.length() })
+    def max = list[0].length()
+    def prefix = ""
+    for(int index = 0; index < max; index++) {
+        def currentChar = list[0][index]
+        for (String item : list) {
+            if (item[index] != currentChar) {
+                for (final def entry in list) {
+                    result.put(entry, entry.substring(prefix.length()))
+                }
+                return result
+            }
```

Comment:
For clarity, I think you could refactor this a bit by making this `if` condition cause a `break` on the outermost loop, and then moving the final iteration that does the prefix-stripping to after the loop.